e2e/nomostest/wait_for_sync.go

```diff
@@ -109,102 +110,102 @@ func syncDirectory(syncDirectoryMap map[types.NamespacedName]string, nn types.Na
 	return AcmeDir
 }
 
-// WaitForRepoSyncs is a convenience method that waits for all repositories
-// to sync.
+// WatchForAllSyncs calls WatchForSync on all Syncs in nt.RootRepos & nt.NonRootRepos.
 //
-// Unless you're testing pre-CSMR functionality related to in-cluster objects,
-// you should be using this function to block on ConfigSync to sync everything.
-//
-// If you want to check the internals of specific objects (e.g. the error field
-// of a RepoSync), use nt.Validate() - possibly in a Retry.
-func (nt *NT) WaitForRepoSyncs(options ...WaitForRepoSyncsOption) {
-	nt.T.Helper()
-
-	waitForRepoSyncsOptions := newWaitForRepoSyncsOptions(nt.DefaultWaitTimeout, DefaultRootSha1Fn, DefaultRepoSha1Fn)
+// If you want to validate specific fields of a Sync object, use
+// nomostest.WatchObject() instead.
+func (nt *NT) WatchForAllSyncs(options ...WatchForAllSyncsOptions) error {
```

Comment:
Whew this signature change leads to quite the diff..

I don't think `Watch` -> `Wait` is necessarily needed in the function name (`watch` is sort of an implementation detail), but I agree with renaming `Repo` -> `All` (`RepoSync` has a specific meaning and this waits for all RSyncs). I also agree with returning the `error` rather than calling `Fatal` in the helper function. So overall I'd say changing the signature is justified, the cost is just a very large diff 😅

Reply:
I've been doing Watch -> Wait for all the conversions, more to check them off the list than because they need to be changed. There's only a few left, but I didn't want to stuff more change into this PR.